src/component/common/ConstraintAccordion/ConstraintAccordionList/ConstraintAccordionList.tsx

```diff
@@ -0,0 +1,142 @@
+import { IConstraint } from 'interfaces/strategy';
+import React, { forwardRef, useImperativeHandle } from 'react';
+import { ConstraintAccordion } from 'component/common/ConstraintAccordion/ConstraintAccordion';
+import produce from 'immer';
+import useUnleashContext from 'hooks/api/getters/useUnleashContext/useUnleashContext';
+import PermissionButton from 'component/common/PermissionButton/PermissionButton';
+import {
+    CREATE_FEATURE_STRATEGY,
+    UPDATE_FEATURE_STRATEGY,
+} from 'component/providers/AccessProvider/permissions';
+import { useWeakMap } from 'hooks/useWeakMap';
+import { objectId } from 'utils/object-id';
+import { useStyles } from './ConstraintAccordionList.styles';
+import { createEmptyConstraint } from 'component/common/ConstraintAccordion/ConstraintAccordionList/createEmptyConstraint';
+import ConditionallyRender from 'component/common/ConditionallyRender';
+
+interface IConstraintAccordionListProps {
+    projectId?: string;
+    environmentId?: string;
+    constraints: IConstraint[];
+    setConstraints: React.Dispatch<React.SetStateAction<IConstraint[]>>;
+    showCreateButton?: boolean;
+}
+
+// Ref methods exposed by this component.
+export interface IConstraintAccordionListRef {
+    addConstraint: (contextName: string) => void;
+}
+
+// Extra form state for each constraint.
+interface IConstraintAccordionListItemState {
+    // Is the constraint currently being edited?
+    editing?: boolean;
+    // Is the constraint new (not yet saved)?
+    unsaved?: boolean;
```

Comment:
Is this true only if the constraint is new or is it also true if the constraint has unsaved changes?

Reply:
Right, this is only `true` when the constraint is brand new (i.e. it has never been saved), used by the cancel button.

Reply:
Got it! I don't know if it needs a better name or not, it just wasn't clear to me when looking at the diff. Maybe `new`/`isNew` would be better? But again, if it's clear from looking at the code, then that's probably enough. An outside view isn't always helpful for things like this 💁🏼

Reply:
Definitely, `new` seems much better. Will fix.